app/validators/allowed_email_domain_validator.rb

```diff
@@ -0,0 +1,17 @@
+class AllowedEmailDomainValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    if value.present?
+      return if value =~ /\.gov\.uk\z/i
+
+      # TODO: we might not want to check against current user domain
+      # when we have a proper allow list?
+      if record.respond_to?(:current_user) && record.current_user.present?
+        user_domain_with_at = record.current_user.email
+          .then { |email| email[email.rindex("@")..] }
+        return if user_domain && value.end_with?(user_domain_with_at)
```

Comment:
This reference to `user_domain` wasn't updated and I think that's what's causing your test failure:
```suggestion
        return if user_domain_with_at && value.end_with?(user_domain_with_at)
```

Reply:
Thank you! Fixed now.